src/Extension/FluentVersionedExtension.php

```diff
@@ -325,6 +325,76 @@ public function existsInLocale($locale = null)
         return $this->isDraftedInLocale($locale) || $this->isPublishedInLocale($locale);
     }
 
+    /**
+     * Check if this record has modifications in this locale
+     * Fluent friendly version of @see Versioned::stagesDiffer()
+     *
+     * @param string|null $locale
+     * @return bool
+     */
+    public function stagesDifferInLocale($locale = null)
+    {
+        $record = $this->owner;
+        $id = $record->ID ?: $record->OldID;
+        $class = get_class($record);
+
+        // Need to check if it's versioned
+        if (!$record->hasExtension(Versioned::class)) {
+            return false;
+        }
+
+        // Need to check that it has stages and is not new
+        if (!$id || !$record->hasStages()) {
+            return false;
+        }
+
+        $locale = $locale ?: $this->getRecordLocale()->Locale;
+
+        // Potentially no Locales have been created in the system yet.
+        if (!$locale) {
+            return false;
+        }
+
+        $versionSuffix = FluentVersionedExtension::SUFFIX_VERSIONS;
+        $baseClass = DataObject::getSchema()->baseDataClass($class);
+        $stageTable = DataObject::getSchema()->tableName($baseClass);
+        $liveTable = $stageTable . $versionSuffix;
+        $stagedTable = $record->getLocalisedTable($stageTable) . $versionSuffix;
+
+        // notes:
+        // VL - Versions localised table
+        // V - Versions table
+        $query = <<<SQL
+SELECT "VL"."Version"
+FROM "$stagedTable" as "VL"
+INNER JOIN "$liveTable" as "V"
+    ON "VL"."RecordID" = "V"."RecordID"
+    AND "VL"."Version" = "V"."Version"
+WHERE "VL"."RecordID" = ?
+AND "VL"."Locale" = ?
+AND "V"."WasPublished" = ?
+ORDER BY "VL"."Version" DESC
+LIMIT 1
+SQL;
```

Comment:
Is raw SQL like this compatible with subsites, or other modules that have low-level ORM manipulations?

Reply:
Might be a good idea to give folks an extension point (or something) so that they can  manipulate this query.

Reply:
This is the same as `Versioned::stagesDiffer()`, raw query, no extension points. I think that extension point is beyond scope of this PR and can be added later as needed.